pkg/scripts/os_test.go

```diff
@@ -0,0 +1,248 @@
+/*
+Copyright 2019 The KubeOne Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scripts
+
+import (
+	"testing"
+
+	"github.com/kubermatic/kubeone/pkg/apis/kubeone"
+	"github.com/kubermatic/kubeone/pkg/testhelper"
+)
+
+func genCluster() kubeone.KubeOneCluster {
+	return kubeone.KubeOneCluster{
+		Versions: kubeone.VersionConfig{
+			Kubernetes: "v1.17.4",
```

Comment:
```suggestion
			Kubernetes: "1.17.4",
```

The leading `v` in the version generates invalid scripts. Do we have some sort of validation or do we drop the leading `v` if the user provides it?

Reply:
kubeone.KubeOneCluster is validated and defaulted of course, long before it get to this point.

Reply:
I've tested this and if the user provides a config file such as the following one, provisioning would fail.

```yaml
apiVersion: kubeone.io/v1alpha1
kind: KubeOneCluster
versions:
  kubernetes: "v1.18.2"
cloudProvider:
  name: aws
```

We should add a validation rule or defaulting to default to the version without the leading `v`. If you prefer, we can do it in a follow-up.

Reply:
heh, yeah, `v, err := semver.NewVersion(version.Kubernetes)` will parse with `v` or without successfully.